### PR TITLE
Plugin draw manager

### PR DESCRIPTION
**Pull request by dnaranjo89**
Makes all the calls with the API except for the favourites management

#### Incomplete stuff
 * The **errors in the forms are not rendered!** (We need a better response)
 * The ***Try* option returns the results with an alert**

### CODE REVIEW

**Comment by Mariocj89** [comment]
Ci failing

**Comment by Mariocj89** [comment]
What happens with the ubchecked checkboxes. Did u check the link i sent u?

**Comment by dnaranjo89** [comment]
Ci fails because the normal draws should be automatically tossed when created
The checkboxes are fine, I don't remember what link you mean

**Comment by Mariocj89** [comment]
Link about getting data from forms.
Can u add the toss yo the test so we  keep the ci passing?

**Comment by Mariocj89** [comment]
!

File: server/forms/form_base.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ class FormBase(forms.Form):
     is_shared = forms.BooleanField(required=False, widget=forms.HiddenInput())
     """Whether the draw is open to multiple users"""
 
+    # TODO the users field is not needed anymore in the form
```
Comment (Mariocj89) [inline server/forms/form_base.py]: we should remove them then.

Comment (Mariocj89) [inline server/forms/form_base.py]: agree we can do it in another PR. but add it as a TODO comment at least and then we can create an issue with all of them once we merge

File: static/js/draw_manager.js [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+                return parseInt(value, 10);
+            }else if (type == "checkbox" ){
+                return this.prop( "checked");
+            }else if (type == "hidden" || type == "radio"){
+                if (value == "True"){
+                    return true;
+                }else if (value == "False"){
+                    return false;
+                }else if (!isNaN(value)){
+                        return parseInt(value, 10);
```
Comment (Mariocj89) [inline static/js/draw_manager.js]: indentation

File: static/js/draw_manager.js [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+            })
+            .filter(function(){
+                var type = this.type;
+                return this.name && $.inArray(this.name, excluded_fields) < 0 &&
+                    !jQuery( this ).is( ":disabled" ) && rsubmittable.test( this.nodeName ) &&
+                    !rsubmitterTypes.test( type ) && ( this.checked || !manipulation_rcheckableType.test( type ) );
+            })
+            .map(function( i, elem ){
+                var value = jQuery( this ).cast_input_value();
+                return value == null ?
```
Comment (Mariocj89) [inline static/js/draw_manager.js]:
```
if (value !== null) {
    return {...}
} else {
    return null
}
```

Comment (Mariocj89) [inline static/js/draw_manager.js]: can it be undefined?

File: static/js/draw_manager.js [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+                    // Set url to the FB share button
+                    $('#share-fb-icon').attr('data-href', url_draw_web);
+                    if (typeof FB !== 'undefined') { //refresh facebook items
+                        FB.XFBML.parse();
+                    }
+
+                    // Set the link of the "Go to the draw" button
+                    $('#go-to-draw').attr('href', url_draw_web);
+
+                    // TODO Show next step in the creation
```
Comment (Mariocj89) [inline static/js/draw_manager.js]: Todo? isnt it done?

File: static/js/draw_manager.js [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+         */
+        create_and_toss: function(){
+            var that = this;
+
+            this.create_draw(
+                callback_done = function( data, textStatus, xhr ){
+                    // Get the url to the draw
+                    var url_draw_api = xhr.getResponseHeader('Location');
+                    var url_draw_web = url_draw_api.replace(/api\/v[\d\.]+\//g,'');
+                    // TODO Could we just auto-toss in normal draws when they are created?
```
Comment (Mariocj89) [inline static/js/draw_manager.js]: that is done

File: static/js/draw_manager.js [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+         * If the draw is created it is automatically tossed and the results are presented
+         * If not, the errors are rendered in the form
+         *
+         * ONLY USED IN NORMAL DRAWS
+         */
+        create_and_toss: function(){
+            var that = this;
+
+            this.create_draw(
+                callback_done = function( data, textStatus, xhr ){
```
Comment (Mariocj89) [inline static/js/draw_manager.js]: dont add it here, but you should really have a look to bluebird

File: static/js/draw_manager.js (hunk 3 above)
Comment (dnaranjo89) [inline static/js/draw_manager.js]: No, it's an input's value so it could only ```""``` or ```null```

File: server/forms/form_base.py (hunk 1 above)
Comment (dnaranjo89) [inline server/forms/form_base.py]: Issue opened #511
